esp/scm/ws_workunits.ecm

```diff
@@ -1690,14 +1690,16 @@ ESPrequest [nil_remove] WUCreateZAPInfoRequest
     string ProblemDescription;
     string WhatChanged;
     string WhereSlow;
-    [min_ver("1.53")] string Password;
+    [depr_ver("1.70")] string Password;
```

Comment:
Didn't notice this before, but this should have both a min_version (when the parameter was added) AND a depr_ver (when the parameter was removed).

Reply:
Added back, but it caused tons of build errors like this:
```
 Building CXX object ecl/eclcmd/CMakeFiles/ecl.dir/__/__/generated/ws_workunits_esp.cpp.o
In file included from /home/ati/HPCC-Platform-build/generated/ws_workunits_esp.ipp:15:0,
                 from /home/ati/HPCC-Platform-build/generated/ws_workunits.esp:7,
                 from /home/ati/HPCC-Platform-build/generated/ws_workunits_esp.cpp:4:
/home/ati/HPCC-Platform-build/generated/ws_workunits.hpp:6711:23: error: ‘virtual const char* IConstWUCreateZAPInfoRequest::getPassword()’ cannot be overloaded
  virtual const char * getPassword()=0;
                       ^
/home/ati/HPCC-Platform-build/generated/ws_workunits.hpp:6710:23: error: with ‘virtual const char* IConstWUCreateZAPInfoRequest::getPassword()’
  virtual const char * getPassword()=0;
                       ^
/home/ati/HPCC-Platform-build/generated/ws_workunits.hpp:6728:15: error: ‘virtual void IEspWUCreateZAPInfoRequest::setPassword(const char*)’ cannot be overloaded
  virtual void setPassword(const char * val)=0;
               ^
/home/ati/HPCC-Platform-build/generated/ws_workunits.hpp:6727:15: error: with ‘virtual void IEspWUCreateZAPInfoRequest::setPassword(const char*)’
  virtual void setPassword(const char * val)=0;
               ^
/home/ati/HPCC-Platform-build/generated/ws_workunits.hpp:6749:15: error: ‘virtual void IClientWUCreateZAPInfoRequest::setPassword(const char*)’ cannot be overloaded
  virtual void setPassword(const char * val)=0;
               ^
/home/ati/HPCC-Platform-build/generated/ws_workunits.hpp:6748:15: error: with ‘virtual void IClientWUCreateZAPInfoRequest::setPassword(const char*)’
  virtual void setPassword(const char * val)=0;
               ^
In file included from /home/ati/HPCC-Platform-build/generated/ws_workunits_esp.ipp:15:0,
                 from /home/ati/HPCC-Platform-build/generated/ws_workunits.esp:7,
                 from /home/ati/HPCC-Platform-build/generated/ws_workunits_esp.cpp:4:
/home/ati/HPCC-Platform-build/generated/ws_workunits.hpp:7847:288: error: redefinition of ‘const char* Password_’
 ged_, const char * WhereSlow_, const char * Password_, const char * Password_, 
                                                                     ^
/home/ati/HPCC-Platform-build/generated/ws_workunits.hpp:7847:264: note: ‘const char* Password_’ previously declared here
 _, const char * WhatChanged_, const char * WhereSlow_, const char * Password_, 
                                                                     ^
In file included from /home/ati/HPCC-Platform-build/generated/ws_workunits.esp:7:0,
                 from /home/ati/HPCC-Platform-build/generated/ws_workunits_esp.cpp:4:
/home/ati/HPCC-Platform-build/generated/ws_workunits_esp.ipp:18412:18: error: redeclaration of ‘SoapStringParam ws_workunits::CWUCreateZAPInfoRequest::m_Password’
  SoapStringParam m_Password;
                  ^
/home/ati/HPCC-Platform-build/generated/ws_workunits_esp.ipp:18411:18: note: previous declaration ‘SoapStringParam ws_workunits::CWUCreateZAPInfoRequest::m_Password’
  SoapStringParam m_Password;
                  ^
/home/ati/HPCC-Platform-build/generated/ws_workunits_esp.ipp:18508:15: error: ‘const char* ws_workunits::CWUCreateZAPInfoRequest::getPassword()’ cannot be overloaded
  const char * getPassword();
               ^
/home/ati/HPCC-Platform-build/generated/ws_workunits_esp.ipp:18507:15: error: with ‘const char* ws_workunits::CWUCreateZAPInfoRequest::getPassword()’
  const char * getPassword();
               ^
/home/ati/HPCC-Platform-build/generated/ws_workunits_esp.ipp:18521:7: error: ‘void ws_workunits::CWUCreateZAPInfoRequest::setPassword(const char*)’ cannot be overloaded
  void setPassword(const char * val);
       ^
/home/ati/HPCC-Platform-build/generated/ws_workunits_esp.ipp:18520:7: error: with ‘void ws_workunits::CWUCreateZAPInfoRequest::setPassword(const char*)’
  void setPassword(const char * val);
       ^
In file included from /home/ati/HPCC-Platform-build/generated/ws_workunits.esp:7:0,
                 from /home/ati/HPCC-Platform-build/generated/ws_workunits_esp.cpp:4:
/home/ati/HPCC-Platform-build/generated/ws_workunits_esp.ipp:20679:288: error: redefinition of ‘const char* Password_’
 ged_, const char * WhereSlow_, const char * Password_, const char * Password_, 
                                                                     ^
/home/ati/HPCC-Platform-build/generated/ws_workunits_esp.ipp:20679:264: note: ‘const char* Password_’ previously declared here
 _, const char * WhatChanged_, const char * WhereSlow_, const char * Password_, 
                                                                     ^
In file included from /home/ati/HPCC-Platform-build/generated/ws_workunits_esp.cpp:4:0:
/home/ati/HPCC-Platform-build/generated/ws_workunits.esp: In constructor ‘ws_workunits::CWUCreateZAPInfoRequest::CWUCreateZAPInfoRequest(const char*, IRpcMessageBinding*)’:
/home/ati/HPCC-Platform-build/generated/ws_workunits.esp:56828:1: error: multiple initializations given for ‘ws_workunits::CWUCreateZAPInfoRequest::m_Password’
 CWUCreateZAPInfoRequest::CWUCreateZAPInfoRequest(const char *serviceName, IRpcM
 ^
/home/ati/HPCC-Platform-build/generated/ws_workunits.esp: In constructor ‘ws_workunits::CWUCreateZAPInfoRequest::CWUCreateZAPInfoRequest(const char*, const char*)’:
/home/ati/HPCC-Platform-build/generated/ws_workunits.esp:56846:1: error: multiple initializations given for ‘ws_workunits::CWUCreateZAPInfoRequest::m_Password’
 CWUCreateZAPInfoRequest::CWUCreateZAPInfoRequest(const char *serviceName, const
 ^
/home/ati/HPCC-Platform-build/generated/ws_workunits.esp: In constructor ‘ws_workunits::CWUCreateZAPInfoRequest::CWUCreateZAPInfoRequest(const char*, IRpcMessage*)’:
/home/ati/HPCC-Platform-build/generated/ws_workunits.esp:56855:1: error: multiple initializations given for ‘ws_workunits::CWUCreateZAPInfoRequest::m_Password’
 CWUCreateZAPInfoRequest::CWUCreateZAPInfoRequest(const char *serviceName, IRpcM
 ^
/home/ati/HPCC-Platform-build/generated/ws_workunits.esp: In constructor ‘ws_workunits::CWUCreateZAPInfoRequest::CWUCreateZAPInfoRequest(IEspContext*, const char*, IProperties*, MapStrToBuf*)’:
/home/ati/HPCC-Platform-build/generated/ws_workunits.esp:56865:1: error: multiple initializations given for ‘ws_workunits::CWUCreateZAPInfoRequest::m_Password’
 CWUCreateZAPInfoRequest::CWUCreateZAPInfoRequest(IEspContext* ctx, const char *
 ^
/home/ati/HPCC-Platform-build/generated/ws_workunits.esp: At global scope:
/home/ati/HPCC-Platform-build/generated/ws_workunits.esp:57331:14: error: redefinition of ‘const char* ws_workunits::CWUCreateZAPInfoRequest::getPassword()’
 const char * CWUCreateZAPInfoRequest::getPassword() { return m_Password.query()
              ^
/home/ati/HPCC-Platform-build/generated/ws_workunits.esp:57330:14: note: ‘virtual const char* ws_workunits::CWUCreateZAPInfoRequest::getPassword()’ previously defined here
 const char * CWUCreateZAPInfoRequest::getPassword() { return m_Password.query()
              ^
/home/ati/HPCC-Platform-build/generated/ws_workunits.esp:57344:6: error: redefinition of ‘void ws_workunits::CWUCreateZAPInfoRequest::setPassword(const char*)’
 void CWUCreateZAPInfoRequest::setPassword(const char * val){ m_Password.set(val
      ^
/home/ati/HPCC-Platform-build/generated/ws_workunits.esp:57343:6: note: ‘virtual void ws_workunits::CWUCreateZAPInfoRequest::setPassword(const char*)’ previously defined here
 void CWUCreateZAPInfoRequest::setPassword(const char * val){ m_Password.set(val
      ^
In file included from /home/ati/HPCC-Platform-build/generated/ws_workunits_esp.cpp:4:0:
/home/ati/HPCC-Platform-build/generated/ws_workunits.esp:70079:299: error: redefinition of ‘const char* Password_’
 ged_, const char * WhereSlow_, const char * Password_, const char * Password_, 
                                                                     ^
/home/ati/HPCC-Platform-build/generated/ws_workunits.esp:70079:275: note: ‘const char* Password_’ previously declared here
 _, const char * WhatChanged_, const char * WhereSlow_, const char * Password_, 
                                                                     ^
ecl/eclcmd/CMakeFiles/ecl.dir/build.make:86: recipe for target 'ecl/eclcmd/CMakeFiles/ecl.dir/__/__/generated/ws_workunits_esp.cpp.o' failed
make[2]: *** [ecl/eclcmd/CMakeFiles/ecl.dir/__/__/generated/ws_workunits_esp.cpp.o] Error 1
CMakeFiles/Makefile2:7827: recipe for target 'ecl/eclcmd/CMakeFiles/ecl.dir/all' failed
```

Reply:
The version declarations are supposed to work together, but let me see if it's broken.

Reply:
I tested some fields with both [min_ver("1.6"), depr_ver("1.7")] and didn't get compile errors.

Reply:
Did you tested it with Password field?